src/PasswordPolicy.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace Drupal\password_enhancements;
+
+use Drupal\password_enhancements\Exception\PolicyInvalidArgumentException;
+use Drupal\user\RoleInterface;
+
+/**
+ * Defines the Password Policy.
+ */
+final class PasswordPolicy implements PasswordPolicyInterface {
```

Comment:
TL;DR; Let's have a call about this class.

Reply:
Pick a date.